pkgs/by-name/py/pyfa/package.nix

```diff
@@ -1,31 +1,124 @@
 {
   lib,
-  appimageTools,
-  fetchurl,
+  python3Packages,
+  fetchFromGitHub,
+  gsettings-desktop-schemas,
+  adwaita-icon-theme,
+  wrapGAppsHook3,
+  gdk-pixbuf,
 }:
 let
-  version = "2.60.2";
+  version = "2.61.0";
 in
-appimageTools.wrapType2 {
+python3Packages.buildPythonApplication rec {
   inherit version;
   pname = "pyfa";
+  format = "other";
 
-  src = fetchurl {
-    name = "pyfa-appimage-${version}";
-    url = "https://github.com/pyfa-org/Pyfa/releases/download/v${version}/pyfa-v${version}-linux.AppImage";
-    hash = "sha256-6doetQ6E1Occ/SqewfxRqPEX1MnUuFomm+8VmetIz4Y=";
+  src = fetchFromGitHub {
+    owner = "pyfa-org";
+    repo = "Pyfa";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-VSuRQugUODc+LbhKbzsA09pnqPEIAt0pQS2An/p7r9A=";
   };
 
+  build-system = [ python3Packages.setuptools ];
+  dependencies = with python3Packages; [
+    wxpython
+    logbook
+    matplotlib
+    python-dateutil
+    requests
+    sqlalchemy_1_4
+    cryptography
+    markdown2
+    beautifulsoup4
+    pyaml
+    roman
+    numpy
+    python-jose
+    requests-cache
+  ];
+
+  buildInputs = [
+    gsettings-desktop-schemas
+    adwaita-icon-theme
+    gdk-pixbuf
+  ];
+
+  dontWrapGApps = true;
+  nativeBuildInputs = [
+    python3Packages.pyinstaller
+    wrapGAppsHook3
+  ];
+
+  #
+  # upstream does not include setup.py
+  #
+  patchPhase = ''
+    cat > setup.py <<EOF
```

Comment:
```suggestion
    cat > setup.py <<"EOF"
```

Reply:
It is currently how it was in the documentation I found regarding no setup.py are you sure this would work? 

I thought the EOF token needs to be by itself

Reply:
this would enter the realm of bash nuance operations

[bash.html#Here-Documents](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Here-Documents
)
> No parameter and variable expansion, command substitution, arithmetic expansion, or filename expansion is performed on word. If any part of word is quoted, the delimiter is the result of quote removal on word, and the lines in the here-document are not expanded. If word is unquoted, all lines of the here-document are subjected to parameter expansion, command substitution, and arithmetic expansion, the character sequence \newline is ignored, and ‘\’ must be used to quote the characters ‘\’, ‘$’, and ‘`’.

sorry, I fail to see the benefit of limiting operations in this case since the unquoted is more trivial / widely used and in this case both yield the same result, can you please detail more situations where using the quoted heredoc would come handy?

Reply:
Removing footguns is clean practice. The fact that here-documents are not commonly quoted in nixpkgs doesn't mean that that is how it should be.

Reply:
Sure, but sorry, gonna have to insist because I still don't see the benefit or find any documentation that justify the clean practice with a good reasoning, can you recomend me any good writing on that?

Reply:
My 2¢ is to follow whatever ShellCheck recommends, and it seems to be completely happy with unquoted heredocs:

![image](https://github.com/user-attachments/assets/ae3f4b2a-bc1c-4993-bbd6-d276c0868ab1)